### ISSUE TITLE
Jump animation wont play any frames but first one

### ISSUE THREAD

**Issue by 2blackbar** [uheartbeast/simple--heart-platformer#1]
Hey i made the jump longer but it just wont play the animation, its stuck at first frame and im not sure what to change to let it play normally, i followed other tuts and jump plays normal but this one is different, its like if animation plauer is dependant on moving or not, strange.
How i can change this ? I want jump to be played when i jump and still play when i release jump.

**Comment by 2blackbar** [comment]
I Fixed it, it was looping on 1st frame which even prevented from playing sound that i added , so i polished it a bit adding more if statements ,  and now jump behaves like it should , heres the code , modified a bit inputs for splitscreen mode:

----------
extends KinematicBody2D

const TARGET_FPS = 60
const ACCELERATION = 8
const MAX_SPEED = 64
const FRICTION = 10
const AIR_RESISTANCE = 1
const GRAVITY = 4
const JUMP_FORCE = 140
export var id = 0
export var speed = 250
var motion = Vector2.ZERO

onready var sprite = $Sprite
onready var animationPlayer = $AnimationPlayer

func _physics_process(delta):
	var x_input = Input.get_action_strength('right_%s' % id) - Input.get_action_strength('left_%s' % id)
	
	if x_input != 0 && is_on_floor():
		animationPlayer.play("Run")
		motion.x += x_input * ACCELERATION * delta * TARGET_FPS
		motion.x = clamp(motion.x, -MAX_SPEED, MAX_SPEED)
		sprite.flip_h = x_input < 0
	if x_input != 0 &&  x_input != 0:
		motion.x += x_input * ACCELERATION * delta * TARGET_FPS
		motion.x = clamp(motion.x, -MAX_SPEED, MAX_SPEED)
		sprite.flip_h = x_input < 0
	if is_on_floor() && x_input == 0 :
		animationPlayer.play("Stand")
	
	motion.y += GRAVITY * delta * TARGET_FPS
	
	if is_on_floor():
		if x_input == 0:
			motion.x = lerp(motion.x, 0, FRICTION * delta)
			
		if Input.is_action_just_pressed('up_%s' % id):
			motion.y = -JUMP_FORCE
			animationPlayer.play("Jump")
	else:
		animationPlayer.play("Jump")
		
		
		if Input.is_action_just_released('up_%s' % id) and motion.y < -JUMP_FORCE/2:
			motion.y = -JUMP_FORCE/2
		
		if x_input == 0:
			motion.x = lerp(motion.x, 0, AIR_RESISTANCE * delta)
	
	motion = move_and_slide(motion, Vector2.UP)

**Comment by uheartbeast** [comment]
Hey! Just saw this :) Glad to hear you were able to figure it out on your
own!

Ben

**Comment by 2blackbar** [comment]
Yeah, Your tutorial with physics is the best one i encountered, do You know where i can find how to do jumpland animation ?
I made this with your player physics tutorial https://www.youtube.com/watch?v=6PfPm0L5aXU

**Comment by uheartbeast** [comment]
Hmmm, I'm not sure I know of a video for that. You could use a new state
for that. Check to see if you are on the ground and if you were in the air
and then enter the land state.
